# Peak at sampled Datasets

## Twitter

In [1]:
from datasets import concatenate_datasets, load_dataset, load_from_disk
import pandas as pd

In [1]:
CORPORA_TWITTER = '/shared/3/projects/hiatus/TOKENIZER_wegmann/data/fitting-corpora/twitter'

In [4]:
train_path = CORPORA_TWITTER
train_data = load_from_disk(train_path)["train"]

Loading dataset from disk:   0%|          | 0/45 [00:00<?, ?it/s]

In [8]:
print(pd.DataFrame(train_data.select(range(10))))

                    id                                               text  \
0  1402478067390550016        @anagodoyxz você sabe quem foi né, que ódio   
1  1429655513126653952  RT @ll9il3: احب الشخص اللي كذا https://t.co/uw...   
2  1411536662203613185  RT @dianaaescorcia: te quiero persona que pone...   
3  1435456553935646724  @aubreystrobel when you put it like that I thi...   
4  1406103063387987971  RT @BRANDRSS: कौन कौन आज न्याय के लिए मांग उठा...   
5  1378558240259153920                  @Sauxy Just say you like corpse🙄🙄   
6  1430017097367756801  RT @t_adabi: ”من مشى لي شبراً، مشيت له ذراعاً....   
7  1468085029138202627                                         ほぼ苦みがないねこれ   
8  1465908754092892161                            https://t.co/V4f3GcsQZM   
9  1349222228248944643                                tenho medo da cindy   

   word_count  
0           8  
1           7  
2          16  
3          12  
4          14  
5           6  
6          14  
7           1  
8       

In [13]:
CORPORA_WIKIPEDIA = "/shared/3/projects/hiatus/TOKENIZER_wegmann/data/fitting-corpora/wikipedia"

In [14]:
wiki_path = CORPORA_WIKIPEDIA
wiki_data = load_from_disk(wiki_path)["train"]

Loading dataset from disk:   0%|          | 0/20 [00:00<?, ?it/s]

In [19]:
wiki_data

Dataset({
    features: ['id', 'text', 'word_count'],
    num_rows: 24877494
})

In [15]:
sum(wiki_data['word_count']) / len(wiki_data['word_count'])

59.08954463017859

In [33]:
pd.set_option('display.max_colwidth', None)  # Show entire content of the column
print(pd.DataFrame(wiki_data.select(range(10))))

       id  \
0  506310   
1  506311   
2  506312   
3  506313   
4  506314   
5  506315   
6  506316   
7  506317   
8  506318   
9  506319   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        

In [2]:
CORPORA_MIXED = '/shared/3/projects/hiatus/TOKENIZER_wegmann/data/fitting-corpora/mixed'
mixed_path = CORPORA_MIXED
mixed_data = load_from_disk(mixed_path)["train"]

Loading dataset from disk:   0%|          | 0/19 [00:00<?, ?it/s]

In [3]:
sum(mixed_data['word_count']) / len(mixed_data['word_count'])

250.3736871861125

In [4]:
# Get unique domains
unique_domains = set(mixed_data['domain'])

In [5]:
# Print an example for each unique domain
for domain in unique_domains:
    # Filter the dataset to get the first occurrence of each domain
    example = mixed_data.filter(lambda x: x['domain'] == domain)[0]
    print(f"Domain: {domain}\nExample Text:\n{example['text']}\nExample Word Count:\n{example['word_count']}\n")

Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: goodreads
Example Text:
This book is terribly written, boring, sexist, homophobic, and super dumb.
Example Word Count:
11


Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: StackExchange
Example Text:
Q:

Android dialog box populate from Database

I want to get list of items inside my dialog box form database. 
I have "mylist" that is on Mainactivity. When I click one list item, I want the lists of 
items inside dialogbox.
final AlertDialog.Builder builder = new AlertDialog.Builder(this);

     ArrayList<String> getdialoglist = new ArrayList<String>();
     final ArrayAdapter<String> adapter1 = new ArrayAdapter<String>(this,
            android.R.layout.simple_list_item_1, getdialoglist);

    ListView dialoglist = (ListView) findViewById(R.id.dialogListView);
    dialoglist.setAdapter(adapter1);

    String[] sColumns = {"_id", "title","description","date","location","trainer_id"};
    final Cursor cursor = db.query("training", sColumns, null, null, null, null, null);

    mylist.setOnItemClickListener(new OnItemClickListener() {

        @Override
        public void onItemClick(AdapterView<?> arg0, View view,
                int position, long 

Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: reddit
Example Text:
It sounds like she has a thing for you and potentially liking your old photos was her way of people flirtatious? When you didn’t say anything back, perhaps she got embarrassed and unfollowed?         One of my favorite sayings is, “life is short, pride be damned...” just go for it and send her a message if you’re interested. You don’t have anything to lose. 
Example Word Count:
63


Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: OpenSubtitles
Example Text:
"Need another bag." " Mike, where you at?" " Yeah, over here." "Drug war?" "Barksdale's coming back on my boy Marlo, huh?" "Shit." "Got it." "Man, I don't know where we at." "We are in Baltimore, Lamar." "Baltimore, Maryland." "That's the second crib of mine them motherfuckers cracked this week." "Stir up a hornet's nest, no telling who gonna get stung." " All paranoid and shit." " Gotta be." "They got Latroy." "Hey, what the fuck is y'all on the streets for, man?" "Don't y'all know there's a fucking war going on, man?" "Get the fuck in the truck, man." "What happened to all them towers?" "Slow train coming." "Reform, Lamar." "Reform." "We should be up on this shit, hearing voices." "We get a signature number on Bodie's phone, we will be." "We got enough PC off the burners alone." "Every time these fucking pen registers click out a call... it screams drug conspiracy." "I'm not the one taking the case to court." "Pearlman wants a specific link to a kn

Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: sfu-socc
Example Text:
I think everyone, including the people of the First Nations, would benefit most if we could put racial differences aside. After all Africa is actually the true home for us all, with the First Nations being the first settlers here, and the Europeans a subsequent set. I do not think it helps to perpetuate divisiveness, and I hope this can be changed for the future of all young people in this great land.
Example Word Count:
73


Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: nytimes-articles-and-comments
Example Text:
<PERSON> 
It would be appropriate at this time to amend Article II of the Constitution. 

In addition to your call for requiring full financial disclosure and addressing conflicts of interest related to business interests, investments et al, presidential nominees should also be vetted by the Defense Counterintelligence and Security Agency. 

Following rigorous background check, nominees must be judged eligible to gain access to specific classified information, perhaps qualify for at least a Top Secret Security level clearance. 

Incredulously, after four years in the position, the current POTUS probably would not qualify to be cleared for even a Confidential level clearance, let alone Sensitive Compartmented Information.
Example Word Count:
105


Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: s2orc
Example Text:

SOME OBSERVATIONS ON THE TOXI- CITY OF EMETINE. in collaboration with In a previous paper(*) we discussed in detail the pharmacological action and therapeutic pro- perties of emetine, and pointed out that it was a potent drug and that its dosage and the length of


m.a., m.d., B.ch. (Cantab.)R N Chopra 
Calcutta School of Tropical Medicine
MAJOR
I.M.S

B N Ghosh 
Calcutta School of Tropical Medicine
MAJOR
I.M.S

Premanicer R C P S ( Glasgow 
Calcutta School of Tropical Medicine
MAJOR
I.M.S

De 
Calcutta School of Tropical Medicine
MAJOR
I.M.S

Cal B ) 
Calcutta School of Tropical Medicine
MAJOR
I.M.S

SOME OBSERVATIONS ON THE TOXI- CITY OF EMETINE. in collaboration with In a previous paper(*) we discussed in detail the pharmacological action and therapeutic pro- perties of emetine, and pointed out that it was a potent drug and that its dosage and the length of



In a previous paper(*) we discussed in detail the pharmacological action and therapeutic proper

Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: blogcorpus
Example Text:
       Today I stopped by the Armory to find out more information about what's going to happen next week. Most of the information is still sketchy, but that's to be expected from the military. I think they should write down information in invisible ink, for all the changes that come along.  I'm also repacking all my gear according to the packing list I received from the company. It's gonna be tough getting all my stuff into duffel bags according to the list, so I'm planning on getting one of my storage bins to put extra stuff in. Tomorrow is the barbecue, so most of my family should be here. I doubt if I'll have a chance to post for a few days, but hopefully I'll be able to write something down early next week.     
Example Word Count:
133


Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: Github
Example Text:
/*
 * Copyright (C) 2010 Francisco Jerez.
 * All Rights Reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining
 * a copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, sublicense, and/or sell copies of the Software, and to
 * permit persons to whom the Software is furnished to do so, subject to
 * the following conditions:
 *
 * The above copyright notice and this permission notice (including the
 * next paragraph) shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
 * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
 * IN NO EVENT SHALL THE COPYRIGHT OWNER(S) AND/OR ITS SUPPLIERS

Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: Gutenberg (PG-19)
Example Text:




Produced by Jeroen Hellingman and the Online Distributed
Proofreading Team at http://www.pgdp.net/ for Project
Gutenberg.








                   The Philippine Islands, 1493-1898

   Explorations by early navigators, descriptions of the islands and
   their peoples, their history and records of the catholic missions,
    as related in contemporaneous books and manuscripts, showing the
   political, economic, commercial and religious conditions of those
   islands from their earliest relations with European nations to the
                    close of the nineteenth century,

                           Volume XXXI, 1640



 Edited and annotated by Emma Helen Blair and James Alexander Robertson
  with historical introduction and additional notes by Edward Gaylord
                                Bourne.


                      The Arthur H. Clark Company
                            Cleveland, Ohio
                                  MCMV








Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: DM Mathematics
Example Text:
1067
List the prime factors of 28798.
2, 7, 11, 17
List the prime factors of 1047.
3, 349
List the prime factors of 2469.
3, 823
List the prime factors of 286.
2, 11, 13
What are the prime factors of 680?
2, 5, 17
What are the prime factors of 27?
3
List the prime factors of 2239.
2239
List the prime factors of 1782.
2, 3, 11
List the prime factors of 11121.
3, 11, 337
What are the prime factors of 48298?
2, 19, 31, 41
List the prime factors of 33350.
2, 5, 23, 29
What are the prime factors of 18449?
19, 971
List the prime factors of 390.
2, 3, 5, 13
List the prime factors of 1208.
2, 151
List the prime factors of 515.
5, 103
What are the prime factors of 326?
2, 163
What are the prime factors of 19445?
5, 3889
List the prime factors of 64391.
19, 3389
List the prime factors of 1020.
2, 3, 5, 17
List the prime factors of 241.
241
What are the prime factors of 13650?
2, 3, 5, 7, 13
List the prime factors of 6142.
2, 37, 83
List the prime factors of 6

Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: gmane
Example Text:
Similar to the comment of <PERSON>, shape expressions could also be a candidate (see http://shex.io).
This can indeed be used as a descriptive language how your data graph would look like, and can be used to test your RDF data (see http://github.com/jvsoest/ShEx). It will show you (from the starting instance/individual) which instances validate or not.

Kind regards,


Example Word Count:
57


Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: amazon
Example Text:
I recently discovered <PERSON> and enjoy her writing style so when I saw this on Netgalley, I couldn't resist. While this novel had a strong start with many elements that make for a deliciously awesome story, it wasn't executed well and dragged in the middle, ultimately lowering the rating.  <PERSON> writes very simply. Her prose isn't flowery or overly complicated. She writes a great story that is easy to read, which is a rare skill and something I appreciate during those times when I just want to escape into a story and forget about the stress I currently have in my life.  In this novel, Lady <PERSON> is currently retired from London as she struggles to learn how to walk again after a debilitating illness. <PERSON> is an impoverished Irish earl who had traveled to England to find a wife whose dowry can hopefully help him save his land and tenants during the potato famine. He has traveled to Lady <PERSON>'s estate at the invitation of her aunt who has agre

Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: realnews
Example Text:
It’s an exciting day for Splatoon 2 fans, as a slew of news regarding the game’s forthcoming updates has been uncovered. Dataminers from the Wii Sports community team have dug up information that indicate the sequel title will introduce new maps and weapons, which appear to include both brand-new additions and returning classics, as well as cosmetics, game modes, and more.
Reddit user <PERSON> the news on the Splatoon 2 forums, thanking the Wii Sports for conducting the mine and then affirming that they are simply the messenger passing on the data. It should be noted that the information collected contains massive spoilers for Splatoon 2, so those wanting to maintain an untainted gameplay experience should view with caution.
The first section of leaked info covers upcoming maps both old and new. In the game’s 1.1.1 update.Incomplete files for two revamped version of the Kelp Dome and Blackbelly Skatepark from the original Splatoon have been introduced. It

Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: bookcorpus
Example Text:
cioè a Marsala. Ma sul più bello gli si rivoltarono le retrovie: liberati momentaneamente dal pericolo punico, i sicelioti ne avevano abbastanza anche di <PERSON>, che dovette tornarsene a Taranto quasi in fuga. Così, nella ennesima oscillazione del pendolo, i cartaginesi finirono nuovamente nei pressi di Siracusa, da cui il neoeletto generale <PERSON> riuscì però a ributtarli. Nel 269 egli fu acclamato re e prese il nome di <PERSON>: l'omonimo predecessore era vissuto due secoli prima e <PERSON> cercava con strani alberi genealogici di dubbia fattura di discenderne per li rami. E già da <PERSON> si chiede a gran voce la guerra contro Cartagine, mentre il grande poeta <PERSON> incita <PERSON> a scacciare dall'isola l'odiato sempiterno nemico, che si è installato perfino a Messina. A questo punto son le cose, quando un terzo interlocutore si fa avanti per dialogare con i due litiganti: <PERSON>.
 Una guarnigione cartaginese, agli ordini di un tale <PERSO

Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: YouTubeCommons
Example Text:
hello what's up internet early morning stream again the last holiday before Oh three minutes and grab it together okay um yeah I usually do these early morning streams because early morning is the only time I get to play now or play with any you know where I I'm sure I won't I won't say get bothered but I won't be disturbed by uh either of my kids needing something and then of course you know you should pay attention to your kids which is why I don't use the game when I know there I know what's up Rafael James so I do really try to play a bit early now so that you know you can get maybe 45 minutes of uninterrupted time although I'm I'm kind of mad Onna sorry I thought it was the skirmish but I'm kind of mad at OverWatch to the way the ranking works now is that you need to do it you get to feeling tight rant it every seven wins or 22 losses so I've been I've had two batches of seven wins like didn't went straight but you know I've won 14 games I've I

Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: ao3
Example Text:
Clarke was told by a short Scottish woman with serious anger management issues (In Clarke's defence she really did look short enough to be a student, and she wouldn't have be that rude to her if she had known that she was the house mother) that she was rooming with the two girls, Octavia and Raven. She couldn't decide whether she liked them or not. The room was nice though, for a dorm room. It was big enough and the wooden floorboards were painted white, gauzy curtains hung over the windows and Octavia seemed to be covering every available surface in fairy lights.
"So that is your bed" Octavia stopped hanging fairy lights around her pin board and pointed to the bed on the far left of the room, underneath the biggest window "there's a loose floorboard underneath if you've got anything to hide."
"Please tell me you're interesting enough to have something to hide new girl" Raven said looking up from the suitcase that she was unpacking.
Clarke raised an eyebrow.
"

Filter:   0%|          | 0/5874119 [00:00<?, ? examples/s]

Domain: Pile-CC
Example Text:
Pop Christmas Market

A 1 day event for mamas, papas, grandparents and anyone else to shop till they drop! We Are Pop, is a carefully curated online and pop up children’s wear store, stocking beautiful pieces from unique, independent brands
Example Word Count:
40


In [6]:
# pd.set_option('display.max_colwidth', None)  # Show entire content of the column
pd.reset_option('display.max_colwidth')
print(pd.DataFrame(mixed_data.select(range(30))))

                                                 text  word_count  \
0   It’s an exciting day for Splatoon 2 fans, as a...         859   
1   It sounds like she has a thing for you and pot...          63   
2   That's the one thing for me that would suck ab...          72   
3   This book is terribly written, boring, sexist,...          11   
4   Similar to the comment of <PERSON>, shape expr...          57   
5   I recently discovered <PERSON> and enjoy her w...         560   
6   Q:\n\nAndroid dialog box populate from Databas...         483   
7   /*\n * Copyright (C) 2010 Francisco Jerez.\n *...         572   
8   Wouldn't it be funny if after all the time the...          59   
9   Q:\n\nsqlsrv_fetch_array() expects parameter 1...         152   
10  Hi everybody,\n\n#1)\nis there any way of gett...          66   
11  Q:\n\nGiven an OracleDataReader, how do I retu...         188   
12  During our read aloud time, my children (ages ...          75   
13  Pop Christmas Market\n\nA 1 da

# Peak at Train Data File

In [3]:
from datasets import concatenate_datasets, load_dataset, load_from_disk
CORPUS_WIKIBOOKS = '/shared/3/projects/hiatus/TOKENIZER_wegmann/data/train-corpora/wikibook'
wikibooks_data = load_from_disk(CORPUS_WIKIBOOKS)["train"]

In [4]:
wikibooks_data

Dataset({
    features: ['id', 'text', 'word_count', 'source'],
    num_rows: 19405169
})

In [6]:
import pandas as pd
print(pd.DataFrame(wikibooks_data.select(range(0, 30))))

          id                                               text  word_count  \
0   40467846  i believed you at first-indeed , i gathered my...          16   
1   12666595       maybe itll work itself out , maybe it wont .          10   
2   10446814                         i 've always been nobody .           6   
3   24378064  his hand deliberately dropped to his immaculat...          17   
4   68503543           terri did n't take the bait to explain .           9   
5   70772620                    her fingers dug into his hair .           7   
6   59568280                                `` no sign of him .           6   
7   20114168  though we were traveling up hill , our momentu...          30   
8   64421744  what he could n't do was keep taking other lov...          25   
9   72553977                          `` just a little bit . ''           7   
10  64157268  his dark , sinful eyes moved from her eyes to ...          18   
11  17361528                           november twen

In [10]:
dataset = wikibooks_data

In [11]:
sum(dataset['word_count']) / len(dataset['word_count'])

49.61202279682851

In [12]:
49 * 19753274

967910426

In [1]:
from datasets import load_from_disk
CORPUS_WEBBOOKS = '/shared/3/projects/hiatus/TOKENIZER_wegmann/data/train-corpora/webbook'
webbook_data = load_from_disk(CORPUS_WEBBOOKS)["train"]
webbook_data

Loading dataset from disk:   0%|          | 0/40 [00:00<?, ?it/s]

Dataset({
    features: ['id', 'text', 'word_count', 'source'],
    num_rows: 6343542
})

In [3]:
import pandas as pd
print(pd.DataFrame(webbook_data.select(range(0, 30))))

                                                   id  \
0   Erika Warmbrunn - Where the Pavement Ends.epub...   
1                                             1232791   
2   An Introduction to Nineteenth-Century Art - Fa...   
3                                            31051637   
4   Tradition and the Black Atlanti - Henry Louis ...   
5        Narrativa - Alejandro Casona.epub.txt_120832   
6                                            28832428   
7                                             5603237   
8                Smoke - Dorianne Laux.epub.txt_12288   
9      Oathbreakers - Mercedes Lackey.epub.txt_116736   
10                                            3629594   
11   The Halloween Tree - Ray Bradbury.epub.txt_13312   
12                                           13091283   
13                                           17544215   
14                                           18396180   
15  Down Amongst the Dead Men (Digital Monda - Ste...   
16                             

In [4]:
# Count the number of occurrences for each source
books3_count = webbook_data.filter(lambda example: example["source"] == "books3").num_rows
openwebtext2_count = webbook_data.filter(lambda example: example["source"] == "OpenWebText2").num_rows

print(f"Books3 Count: {books3_count}")
print(f"OpenWebText2 Count: {openwebtext2_count}")

Filter:   0%|          | 0/6343542 [00:00<?, ? examples/s]

Filter:   0%|          | 0/6343542 [00:00<?, ? examples/s]

Books3 Count: 3185578
OpenWebText2 Count: 3157964


**BookCorpus**

In [7]:
import re

text = "This  is\tan example.\nNewline here."
# This regex will match words and whitespace characters separately
tokens = re.findall(r'\S+|\s+', text)

print(tokens)

['This', '  ', 'is', '\t', 'an', ' ', 'example.', '\n', 'Newline', ' ', 'here.']


### on SADIRI 1%: Do Candidate/Query Overlap: No

In [ ]:
# check the overlap between candidate and query files
from datasets import load_from_disk
directory_path = '/shared/3/projects/hiatus/aggregated_trainset_v2/content_masking_research/down_1_shuffle/train'

In [6]:
# Load the dataset
dataset = load_from_disk(directory_path)

# Display the first few rows of the train split
print(dataset['train'])

Dataset({
    features: ['query_id', 'query_authorID', 'query_text', 'candidate_id', 'candidate_authorID', 'candidate_text'],
    num_rows: 40796
})


In [7]:
# Extract the query_id and candidate_id columns from the train split
query_ids = set(dataset['train']['query_id'])
candidate_ids = set(dataset['train']['candidate_id'])

# Find the intersection of query_id and candidate_id
overlap = query_ids.intersection(candidate_ids)

# Check if there is any overlap
if overlap:
    print("There are overlapping IDs between query_id and candidate_id:")
    print(overlap)
else:
    print("There are no overlapping IDs between query_id and candidate_id.")

There are no overlapping IDs between query_id and candidate_id.


## On SADIRI 100%, amazon: do candidate/query overlap?: No

In [18]:
import pandas as pd
amazon_base = '/shared/3/projects/hiatus/aggregated_trainset_v2/content_masking_research/amazon/'
# Load the train_queries.jsonl file
queries_path = amazon_base + 'train_queries.jsonl'
queries_df = pd.read_json(queries_path, lines=True)

# Load the train_candidates.jsonl file
candidates_path = amazon_base + 'train_candidates.jsonl'
candidates_df = pd.read_json(candidates_path, lines=True)

# Print the keys of the train_candidates.jsonl DataFrame
print("\nKeys in the train_candidates.jsonl DataFrame:")
print(candidates_df.columns)
print(candidates_df[:1])

# Extract query_id and candidate_id from the jsonl datasets
query_ids_jsonl = set(queries_df['documentID'])
candidate_ids_jsonl = set(candidates_df['documentID'])

# Check for overlap between query_id and candidate_id
overlap = query_ids_jsonl.intersection(candidate_ids_jsonl)

if overlap:
    print("There are overlapping IDs between query_id and candidate_id:")
    print(overlap)
else:
    print("There are no overlapping IDs between query_id and candidate_id.")


Keys in the train_candidates.jsonl DataFrame:
Index(['documentID', 'authorIDs', 'fullText', 'spanAttribution', 'isNeedle',
       'collectionNum', 'source', 'dateCollected', 'publiclyAvailable',
       'deidentified', 'languages', 'lengthWords', 'dateCreated',
       'timeCreated', 'sourceSpecific'],
      dtype='object')
                             documentID                 authorIDs  \
0  7041c0a5-6ef6-4e41-8630-5a8409583d89  ['A0103849GBVWICKXD4T6']   

                                            fullText  \
0  Bought this as a possible EDC item, but found ...   

                                     spanAttribution  isNeedle collectionNum  \
0  [{'authorID': 'A0103849GBVWICKXD4T6', 'start':...     False        amazon   

   source dateCollected  publiclyAvailable  deidentified languages  \
0  amazon    2022-12-31               True          True      [en]   

   lengthWords  dateCreated  timeCreated  \
0           72  09 15, 2014  09 15, 2014   

                                

## Get the size of full (not with held-out) SADIRI TRAIN dataset in terms of #words

In [2]:
project_base = "/shared/3/projects/hiatus/aggregated_trainset_v2/content_masking_research/"
data_folders = ["amazon", "ao3", "bookcorpus", "gmane", "nytimes-articles-and-comments", "pubmed", "realnews", "reddit", "stackexchange", "wiki_articles"]
data_folders = [project_base + folder_name for folder_name in data_folders]

In [5]:
import pandas as pd
import os

def count_words_in_fullText(file_path):
    # Load the JSONL file into a DataFrame
    df = pd.read_json(file_path, lines=True)
    
    # Check if the 'fullText' column exists
    if 'fullText' in df.columns:
        # Count the number of words in the 'fullText' column
        word_count = df['fullText'].str.split().str.len().sum()
        return word_count
    else:
        return 0

def process_datasets(dataset_paths):
    total_word_count = 0
    word_counts = {}
    
    for dataset_path in dataset_paths:
        dataset_name = os.path.basename(dataset_path)
        
        # Initialize word count for the dataset
        dataset_word_count = 0
        
        # Process only the specified JSONL files in the dataset folder
        for file_name in ['train_queries.jsonl', 'train_candidates.jsonl']:
            file_path = os.path.join(dataset_path, file_name)
            if os.path.exists(file_path):
                file_word_count = count_words_in_fullText(file_path)
                dataset_word_count += file_word_count
        
        # Store the word count for the dataset
        word_counts[dataset_name] = dataset_word_count
        total_word_count += dataset_word_count
    
    return word_counts, total_word_count

In [33]:
# Process the datasets and get the word counts
word_counts, total_word_count = process_datasets(data_folders)

# Print the word counts for each dataset
for dataset_name, word_count in word_counts.items():
    print(f"Total number of words in {dataset_name}: {word_count}")

# Print the overall total word count
print(f"Overall total number of words: {total_word_count}")

Total number of words in amazon: 31650279
Total number of words in ao3: 573926907
Total number of words in bookcorpus: 57367225
Total number of words in gmane: 141837101
Total number of words in nytimes-articles-and-comments: 24131163
Total number of words in pubmed: 9748317
Total number of words in realnews: 272933709
Total number of words in reddit: 446769021
Total number of words in stackexchange: 153991860
Total number of words in wiki_articles: 34779747
Overall total number of words: 1747135329


In [32]:
word_counts, total_word_count

({'': 34779747}, 1747135329)

### held out SADIRI dataset

In [3]:
data_folders = ["blogcorpus", "food.com-recipes", "sfu-socc", "goodreads", "wiki_discussions"]
data_folders = [project_base + folder_name for folder_name in data_folders]

In [6]:
# Process the datasets and get the word counts
word_counts, total_word_count = process_datasets(data_folders)

# Print the word counts for each dataset
for dataset_name, word_count in word_counts.items():
    print(f"Total number of words in {dataset_name}: {word_count}")

# Print the overall total word count
print(f"Overall total number of words: {total_word_count}")

Total number of words in blogcorpus: 8189607
Total number of words in food.com-recipes: 4346765
Total number of words in sfu-socc: 3007117
Total number of words in goodreads: 53683977
Total number of words in wiki_discussions: 40827678
Overall total number of words: 110055144


## peak at shared/3/datasets/wikipedia/enwiki/pages-articles/enwiki-20230601-pages-articles.clean-text.txt.gz 

In [1]:
wiki_path = "/shared/3/datasets/wikipedia/enwiki/pages-articles/enwiki-20230601-pages-articles.clean-text.txt.gz"

In [2]:
import gzip

def peek_gzipped_file(file_path, num_lines=10):
    with gzip.open(file_path, 'rt', encoding='utf-8') as file:
        for _ in range(num_lines):
            print(next(file))

In [3]:
peek_gzipped_file(wiki_path, num_lines=10)

'Fred' Watson founded the Patent Collapsible Sidecar Company, later renamed to Watsonian Folding Sidecar Company Ltd, after he built a folding sidecar that allowed him to get his motorcycle and sidecar combination through a narrow entrance to the yard at his house.

The first round was held on August 7.

Born in Guildford, Darren is the brother of former England seam bowler Martin Bicknell. However, he struggled to have similar international impact in the 1990s, the selectors instead preferring players such as Jason Gallian and Mark Lathwell. Most of Bicknell's career was spent at Surrey, but he finished his career with seven seasons at Nottinghamshire, before retiring at the end of the 2006 season.

Kairos Na Pame Parakato is a Greek language album by Antonis Remos from 1998 which sold more than 100,000 copies and was certified double platinum."

Thomas Valente (portrayed by Daniel Benzali) was a high-ranking official in the domestic nuclear detection office of the Department of Homel

## peak at /shared/3/datasets/YouTube-Commons  

In [1]:
import dask.dataframe as dd

In [2]:
# Load the dataset using Dask
dataset = dd.read_parquet('/shared/3/datasets/YouTube-Commons')

# Get 10 elements from the dataset
sample = dataset.head(30)

# Display the sample to understand its structure
print(sample)

       video_id                                   video_link  \
0   AjonQW43ZoU  https://www.youtube.com/watch?v=AjonQW43ZoU   
1   TKTxd2icycA  https://www.youtube.com/watch?v=TKTxd2icycA   
2   Bogr9_dWZDY  https://www.youtube.com/watch?v=Bogr9_dWZDY   
3   trkGXvw9KJI  https://www.youtube.com/watch?v=trkGXvw9KJI   
4   ukMZkP-ips0  https://www.youtube.com/watch?v=ukMZkP-ips0   
5   z_l7smopvVg  https://www.youtube.com/watch?v=z_l7smopvVg   
6   pN5EqyT2KNM  https://www.youtube.com/watch?v=pN5EqyT2KNM   
7   8IxzJoOvYXI  https://www.youtube.com/watch?v=8IxzJoOvYXI   
8   JswzVm1akhk  https://www.youtube.com/watch?v=JswzVm1akhk   
9   9_Hun-bAt1o  https://www.youtube.com/watch?v=9_Hun-bAt1o   
10  4CnQdUBRb70  https://www.youtube.com/watch?v=4CnQdUBRb70   
11  PM_O1cxCdxc  https://www.youtube.com/watch?v=PM_O1cxCdxc   
12  kL02ImZMpR4  https://www.youtube.com/watch?v=kL02ImZMpR4   
13  39aTyWu9OzA  https://www.youtube.com/watch?v=39aTyWu9OzA   
14  WahQNXX_0gM  https://www.youtube.com

In [6]:
# Filter the dataset where both transcription_language and original_language are 'en'
filtered_dataset = dataset[(dataset['transcription_language'] == 'en') & (dataset['original_language'] == 'en')]

# Count the number of entries
entry_count = filtered_dataset.shape[0].compute()

# Count unique channel_id
unique_channel_count = filtered_dataset['channel_id'].nunique().compute()

# Sum the word_count
total_word_count = filtered_dataset['word_count'].sum().compute()

# Print the results
print(f"Number of entries: {entry_count}")
print(f"Number of unique channels: {unique_channel_count}")
print(f"Total word count: {total_word_count}")

Number of entries: 1489377
Number of unique channels: 279141
Total word count: 3579987480


## peak at /shared/4/datasets/CommonCrawl/

In [9]:
import lzma
import pandas as pd

# Define the path to your file
file_path = '/shared/4/datasets/CommonCrawl/en.txt.xz'

# Read the .xz file using lzma
with lzma.open(file_path, mode='rt', encoding='utf-8') as file:
    # Read the first 1000 lines (or as many as you need to peek at)
    lines = [next(file) for _ in range(20)]

# Convert the lines to a pandas DataFrame for easier visualization
df = pd.DataFrame(lines)

# Display the first few lines
print(df.head())


                                                   0
0  Belmont Estate is on the market for $63 millio...
1  “Within the city we’ve had homes that have sol...
2  The three-storey home has five bedrooms, twelv...
3  Surrounding the property is a Versailles-inspi...
4  According to Frosch, the listing has received ...


## peak at /shared/4/datasets/thepile

In [13]:
import zstandard as zstd
import json
import os

# Define the directory and file of interest
directory = '/shared/4/datasets/thepile/pile/train'
filename = '00.jsonl.zst'  # Change this to any file you want to peek into

file_path = os.path.join(directory, filename)

# Number of lines to peek at
num_lines_to_peek = 10

# Open the .zst file and read the first few lines
with open(file_path, 'rb') as f:
    dctx = zstd.ZstdDecompressor()
    with dctx.stream_reader(f) as reader:
        buffer = ''
        while num_lines_to_peek > 0:
            chunk = reader.read(1024)  # Read in chunks of 1024 bytes
            if not chunk:
                break
            buffer += chunk.decode('utf-8')
            lines = buffer.split('\n')
            buffer = lines.pop()
            
            for line in lines:
                if num_lines_to_peek == 0:
                    break
                print(json.loads(line))
                num_lines_to_peek -= 1

{'text': 'It is done, and submitted. You can play “Survival of the Tastiest” on Android, and on the web. Playing on the web works, but you have to simulate multi-touch for table moving and that can be a bit confusing.\n\nThere’s a lot I’d like to talk about. I’ll go through every topic, insted of making the typical what went right/wrong list.\n\nConcept\n\nWorking over the theme was probably one of the hardest tasks I had to face.\n\nOriginally, I had an idea of what kind of game I wanted to develop, gameplay wise – something with lots of enemies/actors, simple graphics, maybe set in space, controlled from a top-down view. I was confident I could fit any theme around it.\n\nIn the end, the problem with a theme like “Evolution” in a game is that evolution is unassisted. It happens through several seemingly random mutations over time, with the most apt permutation surviving. This genetic car simulator is, in my opinion, a great example of actual evolution of a species facing a challenge.

In [15]:
filename = '00.jsonl.zst'  # Change this to any file you want to peek into

file_path = os.path.join(directory, filename)

# Number of lines to peek at
num_lines_to_peek = 10000  # Increase the number of lines to ensure we capture all unique pile_set_name

# Set to capture unique pile_set_name values
unique_pile_set_names = set()

# Open the .zst file and read lines
with open(file_path, 'rb') as f:
    dctx = zstd.ZstdDecompressor()
    with dctx.stream_reader(f) as reader:
        buffer = ''
        while num_lines_to_peek > 0:
            chunk = reader.read(1024)  # Read in chunks of 1024 bytes
            if not chunk:
                break
            buffer += chunk.decode('utf-8')
            lines = buffer.split('\n')
            buffer = lines.pop()
            
            for line in lines:
                if num_lines_to_peek == 0:
                    break
                try:
                    data = json.loads(line)
                    if 'meta' in data and 'pile_set_name' in data['meta']:
                        unique_pile_set_names.add(data['meta']['pile_set_name'])
                except json.JSONDecodeError:
                    continue
                num_lines_to_peek -= 1
                
                # Check if we have all unique pile_set_name values
                if len(unique_pile_set_names) == 22:
                    break

# Print the unique pile_set_name values
print(unique_pile_set_names)

{'Ubuntu IRC', 'USPTO Backgrounds', 'NIH ExPorter', 'DM Mathematics', 'HackerNews', 'Gutenberg (PG-19)', 'EuroParl', 'FreeLaw', 'Pile-CC', 'PubMed Central', 'Wikipedia (en)', 'ArXiv', 'OpenSubtitles', 'PubMed Abstracts', 'BookCorpus2', 'YoutubeSubtitles', 'Enron Emails', 'Github', 'PhilPapers', 'Books3', 'StackExchange', 'OpenWebText2'}


## Sample 1.5 billion words from wikipedia

In [8]:
import gzip
import random
from datasets import Dataset
import os

def count_words_in_line(line):
    return len(line.split())

def extract_random_lines(file_path, target_word_count=1_500_000_000, seed=42):
    random.seed(seed)
    total_word_count = 0
    selected_lines = []

    with gzip.open(file_path, 'rt', encoding='utf-8') as file:
        lines = file.readlines()
        random.shuffle(lines)

        for line_number, line in enumerate(lines, 1):
            line_word_count = count_words_in_line(line)
            if total_word_count + line_word_count > target_word_count:
                break
            total_word_count += line_word_count
            selected_lines.append((line_number, line))

    return selected_lines

def save_to_huggingface_format(lines, output_path):
    dataset_dict = {
        "line_number": [line_number for line_number, _ in lines],
        "text": [text for _, text in lines]
    }
    dataset = Dataset.from_dict(dataset_dict)
    dataset.save_to_disk(output_path)

def process_wikipedia_file(input_path, output_path):
    lines = extract_random_lines(input_path)
    save_to_huggingface_format(lines, output_path)

# Replace with your actual file path and desired output path
input_path = "/shared/3/datasets/wikipedia/enwiki/pages-articles/enwiki-20230601-pages-articles.clean-text.txt.gz"
output_path = "/shared/3/projects/hiatus/TOKENIZER_wegmann/data/fitting-corpora/wikipedia"

process_wikipedia_file(input_path, output_path)

Saving the dataset (0/20 shards):   0%|          | 0/25383803 [00:09<?, ? examples/s]

In [5]:
from datasets import load_from_disk

output_path = "/shared/3/projects/hiatus/TOKENIZER_wegmann/data/fitting-corpora/wikipedia"
dataset = load_from_disk(output_path)
first_five_lines = dataset.select(range(5))
for row in first_five_lines:
    print(f"Line {row['line_number']}: {row['text']}")

Loading dataset from disk:   0%|          | 0/20 [00:00<?, ?it/s]

Line 1: Hønefoss in Storelva (Ringerike) valley is left by the line northwest into the Sogna valley, where the route uses its northern valley flanks to Sokna. To get to the Krøderen (lake) valley, Bergen Line follows Rudselva, passes by Langevannet and Breidvannet lakes and also through the 2.3 kilometer Haversting tunnel, which runs parallel to the Norwegian National Road 7, (), Ørgenvik tunnel. Hallingdal valley is used as a natural corridor further north-west, mostly on the left river bank, serving the larger settlements of Flå and Nesbyen. At Svenkerud to the north of Nesbyen the valley flank is changed. Soutwestbound to Gol, the ascent becomes gradually steeper in the following section to Geilo.

Line 2: The Korovin pistol (Пистолет Коровина, Тульский Коровин (ТК), GAU Index 56-A-112) is regarded as the first Soviet semi-automatic pistol.

Line 3: Due to COVID-19 pandemic, this year's Indonesia International Motor Show was split into two events, the IIMS Virtual, which was held vi

## Sample 1.5 billion words from 2021 Twitter API Samples ==> see styletokenizer/create_twitter_sample.py

In [1]:
import bz2
import json
import os
from datasets import Dataset

def count_words(text):
    return len(text.split())

def process_file(file_path, target_word_count_per_file, data):
    cumulative_word_count = 0
    with bz2.open(file_path, 'rt') as file:
        for line in file:
            tweet = json.loads(line)
            text = tweet.get("text", "")
            tweet_id = tweet.get("id", "")
            word_count = count_words(text)
            
            cumulative_word_count += word_count
            data.append({"id": tweet_id, "text": text})
            
            if cumulative_word_count >= target_word_count_per_file:
                return data, cumulative_word_count
    return data, cumulative_word_count


def sample_texts_from_files(directory, target_word_count):
    """
        get the number of compressed files and determine how much to sample from each file,
        st target word count is distributed equally across bz2 files
    """
    files = [os.path.join(root, file_name)
             for root, _, files in os.walk(directory)
             for file_name in sorted(files)
             if not "2022" in file_name and (file_name.endswith('p2.bz2') or file_name.endswith('p1.bz2'))]
    print(files)
    num_files = len(files)
    target_word_count_per_file = target_word_count // num_files
    data = []

    for filen_index, file_path in enumerate(files):
        print(f"At file {filen_index}/{num_files} called {file_path}")
        print(f"Extracting {target_word_count_per_file} words")
        data, _ = process_file(file_path, target_word_count_per_file, data)

    return data

In [2]:
directory = '/nfs/locker/twitter-decahose-locker/2021'
output_path = '/shared/3/projects/hiatus/TOKENIZER_wegmann/data/fitting-corpora/twitter'
target_word_count = 1_500_000_000
data = sample_texts_from_files(directory, target_word_count)
dataset = Dataset.from_dict({"id": [item["id"] for item in data], "text": [item["text"] for item in data]})
dataset.save_to_disk(output_path)

['/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-01.p1.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-01.p2.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-02.p1.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-02.p2.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-03.p1.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-03.p2.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-04.p1.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-04.p2.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-05.p1.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-05.p2.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-06.p1.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-06.p2.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-07.p1.bz2', '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-07.p2.bz2', '/nfs

At file 1/723 called /nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-01.p2.bz2
Extracting 2074688 words
At file 2/723 called /nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-02.p1.bz2
Extracting 2074688 words
At file 3/723 called /nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-02.p2.bz2
Extracting 2074688 words
At file 4/723 called /nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-03.p1.bz2
Extracting 2074688 words
At file 5/723 called /nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-03.p2.bz2
Extracting 2074688 words
At file 6/723 called /nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-04.p1.bz2
Extracting 2074688 words
At file 7/723 called /nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-04.p2.bz2
Extracting 2074688 words
At file 8/723 called /nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-05.p1.bz2
Extracting 2074688 words
At file 9/723 called /nfs/locker/twitter-decahose-locker/2021/decahose.2021-01-05.p2.bz2

KeyboardInterrupt: 

## Sample 1.5 billion for a mixed dataset

SOURCE			GENRE					DOMAIN 								TOTAL 				 SAMPLE


SADIRI			forum					reddit	 						  446,769,021 		   249,000,000
SADIRI			literature				ao3					  			  573,926,907 		   100,000,000
SADIRI			literature				BookCorpus2 					   57,367,225 			50,000,000
ThePile			literature 				Gutenberg before 1919 					?				50,000,000			
SADIRI			news					realnews						  272,933,709 		   169,000,000
SADIRI			news/comments			nytimes-articles-and-comments:     24,131,163 			24,131,163
SADIRI			news/comments			sfu-socc 							3,007,117 			 3,007,117
ThePile			Q&A 					stackexchange 				  	  		? 		   	   200,000,000
SADIRI			reviews					goodreads 					  	   53,683,977  			53,683,977
SADIRI			reviews					amazon	 						   31,650,279 			31,650,279
SADIRI			mails					gmane 						      141,837,101  		   141,837,101
YouTubeCommons	transcripts 			YouTubeCommons 					  	  				   100,000,000
ThePile			transcripts				OpenSubtitles 											50,000,000
ThePile			code				 	Github													50,000,000
WET CommonCrawl	raw text webpages 		CommonCrawl 								    	   100,000,000
s2orc			science					s2orc					  	    				       100,000,000
ThePile			Mathematics 			DM Mathematics  										20,000,000
SADIRI			blogs					blogcorpus	 						 			 		 8,189,607
																		--------------- 	---------------
										TOTAL: 						   		 				 1,500,499,244 

### SADIRI samples

In [2]:
project_base = "/shared/3/projects/hiatus/aggregated_trainset_v2/content_masking_research/"
data_folders = ["reddit", "ao3", "bookcorpus", "realnews", "nytimes-articles-and-comments", "sfu-socc", "goodreads",  "amazon",  "gmane", "blogcorpus"]
data_folders = [project_base + folder_name for folder_name in data_folders]
# word_samples = [249000000, 
#                 100000000, 
#                 50000000, 
#                 169000000, 
#                 24131163, 
#                 3007117,
#                 53683977,
#                 31650279,
#                 141837101,
#                 8189607]
word_samples = [10 for _ in range(len(data_folders))]
assert(len(word_samples) == len(data_folders))

In [11]:
import os
import pandas as pd
from collections import defaultdict

def sample_texts_from_dataframe(data_df, target_word_count):
    sampled_texts = []
    document_ids = []
    current_word_count = 0
    data_df = data_df.sample(frac=1).reset_index(drop=True)  # Shuffle the texts
    for idx, row in data_df.iterrows():
        text = row['fullText']
        words = text.split()
        word_count = len(words)
        sampled_texts.append(text)
        document_ids.append(row['documentID'])
        current_word_count += word_count
        if current_word_count >= target_word_count:
            break
    return {
        "sampled_texts": sampled_texts,
        "sampled_word_count": current_word_count,
        "document_ids": document_ids
    }

def sample_sadiri_texts(dataset_paths, word_samples):
    data_samples = defaultdict(list)
    total_word_count = defaultdict(int)
    
    for dataset_path, word_count in zip(dataset_paths, word_samples):
        dataset_name = os.path.basename(dataset_path)
                
        combined_df = pd.DataFrame()
        
        # Combine both JSONL files into one dataframe
        for file_name in ['train_queries.jsonl', 'train_candidates.jsonl']:
            file_path = os.path.join(dataset_path, file_name)
            if os.path.exists(file_path):
                data_df = pd.read_json(file_path, lines=True)
                print(f"Loaded {len(data_df)} rows from {file_path}")
                combined_df = pd.concat([combined_df, data_df], ignore_index=True)
            else:
                print(f"{file_name} does not exist in {dataset_path}.")
                
        sample_dict = sample_texts_from_dataframe(data_df, word_count)
                
        data_samples['source'] += ["SADIRI" for _ in range(len(sample_dict))]
        data_samples['domain'] += [dataset_name for _ in range(len(sample_dict))]
        data_samples['text'] += sample_dict["sampled_texts"]
        data_samples['id'] += sample_dict["document_ids"]
        total_word_count[dataset_name] += sample_dict["sampled_word_count"]
        print(f"Total words collected for {dataset_name}: {total_word_count[dataset_name]}")
      

    print(data_samples)

In [12]:
sample_sadiri_texts(data_folders, word_samples)

Loaded 2090161 rows from /shared/3/projects/hiatus/aggregated_trainset_v2/content_masking_research/reddit/train_queries.jsonl
Loaded 2090161 rows from /shared/3/projects/hiatus/aggregated_trainset_v2/content_masking_research/reddit/train_candidates.jsonl
Total words collected for reddit: 74
Loaded 317274 rows from /shared/3/projects/hiatus/aggregated_trainset_v2/content_masking_research/ao3/train_queries.jsonl
Loaded 317274 rows from /shared/3/projects/hiatus/aggregated_trainset_v2/content_masking_research/ao3/train_candidates.jsonl
Total words collected for ao3: 544
Loaded 56319 rows from /shared/3/projects/hiatus/aggregated_trainset_v2/content_masking_research/bookcorpus/train_queries.jsonl
Loaded 56319 rows from /shared/3/projects/hiatus/aggregated_trainset_v2/content_masking_research/bookcorpus/train_candidates.jsonl
Total words collected for bookcorpus: 512
Loaded 181024 rows from /shared/3/projects/hiatus/aggregated_trainset_v2/content_masking_research/realnews/train_queries.json

###  SKIPPING as it is unclear what en.txt.xz is -- WET Common Crawl dataset

In [ ]:
wet_file_path = '/shared/4/datasets/CommonCrawl/en.txt.xz'

In [45]:
import lzma
import random

def estimate_words_per_line(file_path, sample_size=1000):
    total_lines = 0
    sampled_words = 0
    with lzma.open(file_path, mode='rt', encoding='utf-8') as file:
        for line in file:
            total_lines += 1
            if total_lines <= sample_size:
                sampled_words += len(line.split())
    average_words_per_line = sampled_words / sample_size
    return total_lines, average_words_per_line

def sample_common_crawl(dataset_paths, word_samples):
    sampled_lines = []
    line_numbers = []
    total_current_word_count = 0
    
    # First pass to estimate total number of lines and average words per line
    total_lines, average_words_per_line = estimate_words_per_line(dataset_paths)
    
    print(total_lines)
    
    # Estimate total word count
    estimated_total_word_count = average_words_per_line * total_lines
    
    # Calculate sampling probability
    sampling_probability = word_samples / estimated_total_word_count
    
    # Second pass to perform sampling
    with lzma.open(dataset_paths, mode='rt', encoding='utf-8') as file:
        line_number = 0
        for line in file:
            line_number += 1
            word_count = len(line.split())
            
            # Decide whether to include this line based on the calculated probability
            if random.random() < sampling_probability:
                sampled_lines.append(line.strip())
                line_numbers.append(line_number)
                total_current_word_count += word_count
                
            # Stop if we've reached the target word count
            if total_current_word_count >= word_samples:
                break
    
    # Construct the result dictionary
    result = {
        "id": line_numbers,
        "text": sampled_lines,
        "domain": ["commoncrawl"] * len(sampled_lines),
        "source": ["commoncrawl"] * len(sampled_lines)
    }
    
    return result

In [46]:
dataset_path = '/shared/4/datasets/CommonCrawl/en.txt.xz'
word_samples = 10
sampled_data = sample_common_crawl(dataset_path, word_samples)
print(sampled_data)

KeyboardInterrupt: 

### ThePile Samples (the pile is already shuffled, see https://arxiv.org/pdf/2101.00027)

In [21]:
pile_set_names = ['Gutenberg (PG-19)', 'StackExchange', 'OpenSubtitles', 'Github', 'Pile-CC' ,'DM Mathematics']
# word_counts = [50000000, 200000000, 50000000, 50000000, 100000000, 20000000]
word_counts = [10 for _ in range(len(pile_set_names))]

In [43]:
import os
import zstandard as zstd
import json

def read_lines_from_zst(file_path):
    """Generator to read lines from a .zst compressed file"""
    with open(file_path, 'rb') as f:
        dctx = zstd.ZstdDecompressor()
        with dctx.stream_reader(f) as reader:
            buffer = ''
            while True:
                chunk = reader.read(1024)  # Read in chunks of 1024 bytes
                if not chunk:
                    break
                buffer += chunk.decode('utf-8')
                lines = buffer.split('\n')
                buffer = lines.pop()
                for line in lines:
                    yield line

def sample_pile(pile_set_names, word_counts):
    dir_path = "/shared/4/datasets/thepile/pile/train"
    zst_files = [f for f in os.listdir(dir_path) if f.endswith('.jsonl.zst')]
    
    sampled_lines = []
    sampled_texts = []
    domains = []

    # Dictionary to keep track of the word count for each pile set name
    word_counts_dict = dict(zip(pile_set_names, word_counts))
    current_word_counts = {name: 0 for name in pile_set_names}
    
    line_counter = 0

    def should_continue_sampling():
        """Check if we need to continue sampling, so ANY requirement not yet fulfilled"""
        return any(current_word_counts[name] < word_counts_dict[name] for name in pile_set_names)
    
    for filename in zst_files:
        if not should_continue_sampling():
            break

        
        file_path = os.path.join(dir_path, filename)
        for line in read_lines_from_zst(file_path):
            if not should_continue_sampling():
                break

            try:
                data = json.loads(line)
                pile_set_name = data.get('meta', {}).get('pile_set_name')
                if (pile_set_name in pile_set_names) and (current_word_counts[pile_set_name]<word_counts_dict[pile_set_name]):
                    text = data.get('text', '')
                    text_word_count = len(text.split())
                    sampled_lines.append(line_counter)
                    sampled_texts.append(text)
                    domains.append(pile_set_name)
                    current_word_counts[pile_set_name] += text_word_count
            except json.JSONDecodeError:
                print("decode error")
                continue
            line_counter += 1

    # Return the sampled data
    return {
        "id": sampled_lines,
        "domain": domains,
        "source": ["thePile"] * len(sampled_texts),
        "text": sampled_texts,
    }


In [44]:
result = sample_pile(pile_set_names, word_counts)
print(result)

{'id': [1, 19, 62, 257, 3421], 'domain': ['StackExchange', 'Github', 'DM Mathematics', 'OpenSubtitles', 'Gutenberg (PG-19)'], 'source': ['thePile', 'thePile', 'thePile', 'thePile', 'thePile'], 'text': ['Q:\n\nWhat is the meaning of 么?\n\nIf you want to say "This dog," you\'d say, "这只狗". If you want to say "This thing," you\'d say, "这个东西". If you want to say, "This much," you\'d say, "这么多".\nI understand the reasons for the 只 and the 个, but what does the 么 really mean?\n\nA:\n\n么 has lost most of its lexical meaning in modern usage, but it appears to have originally been derived from the word 物, meaning \'thing\'. Chinese by Jerry Norman, pp. 119-20 makes mention of the etymology of various modern words featuring 么:\n\nIn an insightful article, Zhāng Hùiyīng (1982) has shown that shénme 什么, the standard modern word for \'what\', comes from shí wù 什物 (MC źjəp mjuət). This form (in a number of different phonological shapes) makes its first appearance in the Tang dynasty. Shí wù 什物 origina

### YouTube Commons Sample

In [1]:
from dask.distributed import Client
import dask.dataframe as dd

def sample_YouTubeCommons(required_word_count):    
    # Load the dataset using Dask
    dataset = dd.read_parquet('/shared/3/datasets/YouTube-Commons')
     print(f"Dataset loaded")

    # Filter the dataset where both transcription_language and original_language are 'en'
    filtered_dataset = dataset[(dataset['transcription_language'] == 'en') & (dataset['original_language'] == 'en')]
    print(f"Dataset filtered")
    
    # Debugging step: Check the number of rows after filtering
    filtered_count = filtered_dataset.shape[0].compute()
    print(f"Number of rows after filtering: {filtered_count}")

    if filtered_count == 0:
        raise ValueError("No data found after filtering. Check the filter conditions or data content.")

    # Shuffle the dataset
    shuffled_dataset = filtered_dataset.sample(frac=1, random_state=42)

    # Compute the shuffled dataset
    filtered_df = shuffled_dataset.compute()

    # Initialize variables to store the sampled texts and the accumulated word count
    sampled_texts = []
    total_word_count = 0
    sampled_ids = []

    # Iterate through the shuffled dataframe and sample texts until the required word count is reached
    for _, row in filtered_df.iterrows():
        sampled_texts.append(row['text'])
        total_word_count += row['word_count']
        sampled_ids.append(row['video_id'])
        if total_word_count >= required_word_count:
            break
    
    # Ensure that we have sampled enough texts to meet the word count requirement
    if total_word_count < required_word_count:
        raise ValueError("Not enough data to meet the required word count")
    
    return {
        "id": sampled_ids,
        "domain": ["YouTubeCommons"] * len(sampled_texts),
        "source": ["YouTubeCommons"] * len(sampled_texts),
        "text": sampled_texts,
    }



In [2]:
# Example usage
try:
    sampled_data = sample_YouTubeCommons(100)
    print(sampled_data)
except Exception as e:
    print(f"Error: {e}")

Number of rows after filtering: 1489377
{'id': ['UJiOpInAP4U'], 'domain': ['YouTubeCommons'], 'source': ['YouTubeCommons'], 'text': ["tonight live from the scullery theater from the impersonator capital of the world in the heart of downtown las vegas we present the downtown podcast starring your host mr tonight's guest from rat pack las vegas peter pavone and from real gaming lawrence vaughn and now ladies and gentlemen gentlemen let's give it up for the man who doesn't mean jerry seinfeld thank you thank you guys oh that was good yeah jerry scifield you already knew jerry what's to do with all the sniffles we're doing a podcast it's not a late night show we got late night show here we got a podcast here it's the audio it's a radio what is it what's the dude i don't know sorry all right i tried it tried we have an impersonator on tonight and i wanted to impress him i don't think i did nope doesn't look impressed okay so my favorite thing happened this last week uh there was a flight fr

### s2orc sample

In [3]:
import os
import json
import random

s2orc_path = "/shared/3/projects/citation-context/s2orc/s2orc"

def count_words(text):
    return len(text.split())

def read_files_and_sample(path, target_word_count):
    word_count = 0
    sampled_ids = []
    sampled_texts = []
    
    files = [os.path.join(path, f) for f in os.listdir(path) if f.startswith('s2orc_')]
    num_files = len(files)
    words_per_file = target_word_count // num_files
    
    for file_path in files:
        with open(file_path, 'r') as f:
            lines = f.readlines()
            total_lines = len(lines)
            random_indices = random.sample(range(total_lines), total_lines)  # Shuffle the line indices
            
            for idx in random_indices:
                if word_count >= target_word_count:
                    break
                
                line = json.loads(lines[idx])
                corpusid = line.get('corpusid')
                text = line.get('content', {}).get('text', "")
                words = text.split()
                
                if word_count + len(words) > target_word_count:
                    remaining_words = target_word_count - word_count
                    sampled_texts.append(" ".join(words[:remaining_words]))
                    sampled_ids.append(corpusid)
                    word_count += remaining_words
                    break
                else:
                    sampled_texts.append(text)
                    sampled_ids.append(corpusid)
                    word_count += len(words)
                
                if word_count >= words_per_file * (files.index(file_path) + 1):
                    break

        if word_count >= target_word_count:
            break

    return sampled_ids, sampled_texts

def sample_s2orc(required_word_count):
    sampled_ids, sampled_texts = read_files_and_sample(s2orc_path, required_word_count)
    return {
        "id": sampled_ids,
        "domain": ["s2orc"] * len(sampled_texts),
        "source": ["s2orc"] * len(sampled_texts),
        "text": sampled_texts,
    }


SyntaxError: incomplete input (2723164150.py, line 1)

In [ ]:
sampled_data = sample_s2orc(10)
print(sampled_data)

## Stats on fitting corpora

In [5]:
from datasets import load_from_disk

# Load the dataset from the specified path with train/dev/test splits
dataset_path = "/shared/3/projects/hiatus/TOKENIZER_wegmann/data/fitting-corpora/wikipedia"
dataset = load_from_disk(dataset_path)

Loading dataset from disk:   0%|          | 0/20 [00:00<?, ?it/s]

In [9]:
train_dataset = dataset['train']
dev_dataset = dataset['dev']
test_dataset = dataset['test']

In [10]:
    print(f"Train word count: {sum(train_dataset['word_count'])}")
    print(f"Dev word count: {sum(dev_dataset['word_count'])}")
    print(f"Test word count: {sum(test_dataset['word_count'])}")

Train word count: 1469999792
Dev word count: 15000029
Test word count: 15000087


In [12]:
if "domain" in train_dataset[0]:
    for split in [train_dataset, dev_dataset, test_dataset]:
        domain_word_count = {}
        for domain, word_count in zip(split["domain"], split["word_count"]):
            if domain not in domain_word_count:
                domain_word_count[domain] = 0
            domain_word_count[domain] += word_count
        print(f"Domain word count distribution for {split}: {domain_word_count}")

            

In [18]:
def count_words(text):
    return len(text.split())

In [1]:
import bz2
import json
data = []
file_path = '/nfs/locker/twitter-decahose-locker/2021/decahose.2021-05-18.p2.bz2'
total_word_count = 0
with bz2.open(file_path, 'rt') as file:
    for line in file:
        tweet = json.loads(line)
        print(tweet)
        break
        text = tweet.get("text", "")
        tweet_id = tweet.get("id", "")
        word_count = count_words(text)
        data.append({"id": tweet_id, "text": text, "word_count": word_count})
        total_word_count += word_count
        if total_word_count > 3000000:
            break
# print(data)

{'created_at': 'Tue May 18 14:27:16 +0000 2021', 'id': 1394660866600013825, 'id_str': '1394660866600013825', 'text': 'RT @kelvinstych: zenti, i have something called ultra thin condom. wanna try? please.......', 'source': '<a href="http://twitter.com/download/android" rel="nofollow">Twitter for Android</a>', 'truncated': False, 'in_reply_to_status_id': None, 'in_reply_to_status_id_str': None, 'in_reply_to_user_id': None, 'in_reply_to_user_id_str': None, 'in_reply_to_screen_name': None, 'user': {'id': 2543444227, 'id_str': '2543444227', 'name': '˚ ༘ บู้เอง ₊˚ˑ', 'screen_name': 'bubu7910', 'location': None, 'url': None, 'description': None, 'translator_type': 'none', 'protected': False, 'verified': False, 'followers_count': 105, 'friends_count': 191, 'listed_count': 3, 'favourites_count': 12339, 'statuses_count': 133839, 'created_at': 'Tue Jun 03 12:41:48 +0000 2014', 'utc_offset': None, 'time_zone': None, 'geo_enabled': False, 'lang': None, 'contributors_enabled': False, 'is_translator'

### Book3Corpus

In [1]:
from datasets import load_from_disk

# Load the dataset from the specified path with train/dev/test splits
dataset_path = "/shared/3/projects/hiatus/aggregated-style-data/book3corpus/train_dataset"
dataset = load_from_disk(dataset_path)

Loading dataset from disk:   0%|          | 0/22 [00:00<?, ?it/s]

In [2]:
import pandas as pd
print(pd.DataFrame(dataset.select(range(10, 20))))

                             documentID  \
0  13269867-fa7c-a20e-706f-5f374fa23578   
1  a83e8b0e-01a0-a2da-6fb9-9a30ae8a12eb   
2  1c72b3c8-c7a0-c0d4-33aa-0c6ffb951a15   
3  2cee9733-9ac3-a90d-343b-c019f5c40853   
4  948431e4-402b-6b1e-3424-8d8a693a78a6   
5  e6a0802c-ddf9-b479-8a7d-703d11326c5c   
6  35c910ad-70a2-b94b-31c6-dc9564c19c17   
7  94b66ac2-1bb1-f44f-0bf3-a14b37143e9a   
8  90148df2-85c8-5aa0-d57d-39010bcaf010   
9  7f59b8d8-3c9b-53c1-26eb-661dd390aebf   

                                authorIDs  \
0  [a623652e-5995-c9da-386c-a48e077d255f]   
1  [a623652e-5995-c9da-386c-a48e077d255f]   
2  [a623652e-5995-c9da-386c-a48e077d255f]   
3  [a623652e-5995-c9da-386c-a48e077d255f]   
4  [a623652e-5995-c9da-386c-a48e077d255f]   
5  [a623652e-5995-c9da-386c-a48e077d255f]   
6  [a623652e-5995-c9da-386c-a48e077d255f]   
7  [a623652e-5995-c9da-386c-a48e077d255f]   
8  [a623652e-5995-c9da-386c-a48e077d255f]   
9  [a623652e-5995-c9da-386c-a48e077d255f]   

                              

### The Pile

In [15]:
import os
import zstandard as zstd
import json
dir_path = "/shared/4/datasets/thepile/pile/train"
def read_lines_from_zst(file_path):
    """Generator to read lines from a .zst compressed file"""
    with open(file_path, 'rb') as f:
        dctx = zstd.ZstdDecompressor()
        with dctx.stream_reader(f) as reader:
            buffer = ''
            while True:
                chunk = reader.read(1024)  # Read in chunks of 1024 bytes
                if not chunk:
                    break
                buffer += chunk.decode('utf-8')
                lines = buffer.split('\n')
                buffer = lines.pop()
                for line in lines:
                    yield line

In [16]:
zst_files = [f for f in os.listdir(dir_path) if f.endswith('.jsonl.zst')]

In [9]:
file_path = os.path.join(dir_path, zst_files[0])

In [17]:
cntr = 0
for line in read_lines_from_zst(file_path):
    data = json.loads(line)
    pile_set_name = data.get('meta', {}).get('pile_set_name')
    if pile_set_name != "OpenWebText2":
        continue
    print(data)
    cntr += 1
    if cntr > 20:
        break

{'text': "There are three developers working on the Dero Project who have been working in the field of cryptography and blockchain development as a team for more than a decade. Dero's research and development team are devoted full time to the project, and can often be found in our Slack channel. For now, the development team are choosing to retain their privacy until the Dero Project incorporates.\n\nDERO will be the first CryptoNote blockchain to have 500 transactions per second on its native blockchain without any lightning networks, validators or off-blockchain solutions\n\nThe Dero Project has written a unique new blockchain technology that is based on the CryptoNote protocol. Dero's goal is to create a unique state of the art blockchain technology with enhanced reliability, privacy, security, usability, and portability by bringing together some of the best proven technologies like the CryptoNote protocol and smart contracts, thereby allowing for the creation of truly private smart

{'text': 'perl 5.20.0 is now available\n\nFrom: Ricardo Signes\n\nDate: May 27, 2014 01:37\n\nSubject: perl 5.20.0 is now available\n\nMessage ID: 20140527013737.GA21381@cancer.codesimply.com\n\nMay 27, 2014 01:37perl 5.20.0 is now available\n\nBut thy eternal summer shall not fade, Nor lose possession of that fair thou ow\'st; Nor shall Death brag thou wander\'st in his shade, When in eternal lines to time thou grow\'st: So long as men can breathe or eyes can see, So long lives this, and this gives life to thee. -- William Shakespeare, Sonnet 18 We are delighted to announce perl v5.20.0, the first stable release of version 20 of Perl 5. You will soon be able to download Perl 5.20.0 from your favorite CPAN mirror or find it at: https://metacpan.org/release/RJBS/perl-5.20.0/ SHA1 digests for this release are: e925e4fc36e90eace19a1ca850f912618ba6788f perl-5.20.0.tar.bz2 52087047fd35e1020387f7c492f594a397be4e12 perl-5.20.0.tar.gz You can find a full list of changes in the file "perldelta.

In [20]:
from langdetect import detect

text = "This is a sample text."
language = detect(text)

In [22]:
text = "This is a sample text."
language = detect(text)